internal/cloud/backend_apply.go

```diff
@@ -182,7 +182,7 @@ func (b *Cloud) opApply(stopCtx, cancelCtx context.Context, op *backendrun.Opera
 	}
 
 	// Do the apply!
-	if !op.AutoApprove && err != errRunApproved {
+	if (!op.AutoApprove || hasSavePlanFile) && err != errRunApproved {
```

Comment:
May be worth adding a comment here for clarity.

Reply:
Good call, included 👍